src/components/Bird/Bird.js

```diff
@@ -24,20 +24,26 @@ const Bird = ({ commonName, scientificName, id }) => {
     return photoURL;
   }
 
-  getImages(scientificName, 1)
-  .then(data => {
-    if (!photos.length) {
-      const url = createURL(data.photos.photo[0])
-      setPhotos(url)
-    }
-  })
-  .catch(() => setError("Please try again later!"))
-  ;
+  useEffect(() => {
```

Comment:
Looks like this fixes the memory leak for Bird List, but not for bird details. Keep slaving over that hot keyword.